src/main/java/org/alfresco/repo/security/permissions/impl/acegi/ACLEntryAfterInvocationProvider.java

```diff
@@ -573,244 +554,194 @@ private ResultSet decideNew(Authentication authentication, Object object, Config
             maxCheckTime = returnedObject.getResultSetMetaData().getSearchParameters().getMaxPermissionCheckTimeMillis();
         }
 
-        if (supportedDefinitions.size() == 0)
-        {
-            if (maxSize == null)
-            {
-                return returnedObject;
-            }
-            else if (returnedObject.length() > maxSize.intValue())
-            {
-                for (int i = 0; i < maxSize.intValue(); i++)
-                {
-                    filteringResultSet.setIncluded(i, true);
-                }
-                filteringResultSet.setResultSetMetaData(new SimpleResultSetMetaData(LimitBy.FINAL_SIZE, PermissionEvaluationMode.EAGER, returnedObject.getResultSetMetaData()
-                        .getSearchParameters()));
-            }
-            else
-            {
-                for (int i = 0; i < maxSize.intValue(); i++)
-                {
-                    filteringResultSet.setIncluded(i, true);
-                }
-                filteringResultSet.setResultSetMetaData(new SimpleResultSetMetaData(LimitBy.UNLIMITED, PermissionEvaluationMode.EAGER, returnedObject.getResultSetMetaData()
-                        .getSearchParameters()));
-            }
-        }
-        if (returnedObject.length() > 0)
-        {
-            //force prefetch before starting record time
-            boolean builkFetch = returnedObject.getBulkFetch();
-            returnedObject.setBulkFetch(false);
-            returnedObject.getNodeRef(returnedObject.length() - 1);
-            returnedObject.setBulkFetch(builkFetch);
-        }
+        FilteringResultSet filteringResultSet = new FilteringResultSet(returnedObject);
 
         // record the start time

```

Comment:
What happens if we exceed the maximum allowed execution time?  Does the user get nothing?